src/web/aws/authentication/AWSCustomEndpoints.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import { ExternalLink } from 'components/common';
+
+import { FormDataContext } from 'aws/context/FormData';
+import { AdvancedOptionsContext } from 'aws/context/AdvancedOptions';
+import AdditionalFields from 'aws/common/AdditionalFields';
+import ValidatedInput from 'aws/common/ValidatedInput';
+import { SectionTitle, SectionNote } from 'aws/common/sharedStyles';
+
+const INPUT_PATTERN = 'https://(.*)';
+const INPUT_MESSAGE = "Domain should begin with 'https://'";
+const INPUT_PLACEHOLDER = 'https://vpce-1234.service.region.vpce.amazonaws.com';
+
+const StyledAdditionalFields = styled(AdditionalFields)`
+  margin: 0 0 35px;
+`;
+
+const StyledSectionTitle = styled(SectionTitle)`
+  margin: 12px 0 0;
+`;
+
+const StyledSectionNote = styled(SectionNote)`
+  margin: 0 0 12px;
+`;
+
+const AWSCustomEndpoints = ({ onChange }) => {
+  const { formData } = useContext(FormDataContext);
+  const { isAWSCustomEndpointsVisible, setAWSCustomEndpointsVisibility } = useContext(AdvancedOptionsContext);
+  const {
+    awsEndpointCloudWatch,
+    awsEndpointDynamoDB,
+    awsEndpointIAM,
+    awsEndpointKinesis,
+  } = formData;
+
+  const handleToggle = (visible) => {
+    setAWSCustomEndpointsVisibility(visible);
+  };
+
+  return (
+    <StyledAdditionalFields title="Optional AWS VPC Endpoints"
+                            visible={isAWSCustomEndpointsVisible}
+                            onToggle={handleToggle}>
+
+      <StyledSectionTitle>Overrides the default AWS API endpoint URL that Graylog communicates with.</StyledSectionTitle>
+      <StyledSectionNote>Only specify these if you are using <ExternalLink href="https://docs.aws.amazon.com/vpc/latest/userguide/vpc-endpoints.html">VPC Endpoints</ExternalLink> for AWS services.</StyledSectionNote>
+
+      <ValidatedInput id="awsEndpointCloudWatch"
```

Comment:
@kyleknighted Are these inputs derived from the server side or they purely on the client side ?

Reply:
@lingpri The value for these input fields is ultimately saved with the input that gets saved in Graylog. These input fields here are for the setup process, which uses the custom UI:

![image](https://user-images.githubusercontent.com/3423655/70981766-1237b980-20b6-11ea-8be0-b99de911323e.png)

 The  input fields defined in [AWSInput](https://github.com/Graylog2/graylog-plugin-integrations/blob/628f89c10ade31d2de5bde1a8a53b19c731cb141/src/main/java/org/graylog/integrations/aws/inputs/AWSInput.java#L153) in Java code, are displayed in the Edit Input page modal dialog:

![image](https://user-images.githubusercontent.com/3423655/70981840-3b584a00-20b6-11ea-93ff-013f9c2c1cac.png)

I hope this helps. Please let me know if you have any other questions.